File: chapter1/section2/ride/ride.java
/*
ID: waz
LANG: JAVA
TASK: ride
*/
import java.io.*;
import java.util.*;

class ride { // The class name must be same as the TASK name!!
  
   public static int readNumber(String s) {
    int ans = 1;
        for (int i = 0; i < s.length(); i++) {
            ans *= (s.charAt(i) - 'A') + 1;
        }
      return ans%47;
  }

  public static void main (String [] args) throws IOException {
    BufferedReader in = new BufferedReader(new FileReader("ride.in"));
    PrintWriter out = new PrintWriter(new BufferedWriter(new FileWriter("ride.out")));
    if(readNumber(in.readLine()) == readNumber(in.readLine())) 
        out.println("GO");
    else
        out.println("STAY");
    in.close();                           
    out.close();                                  
  }
}

File: chapter1/section2/ride/ride.py
"""
ID: waz
LANG: PYTHON2
TASK: ride
"""

def read_number(s) :
	ans = 1
	for i in s:
		ans = ans * (ord(i) - ord('A')+1)
 
	return ans % 47

fin = open('ride.in','r')
fout = open('ride.out','w')

if read_number(fin.readline().strip()) == read_number(fin.readline().strip()):
	fout.write("GO\n")
else :
	fout.write("STAY\n")

fin.close()
fout.close()
